endpoint.go

```diff
@@ -615,16 +637,29 @@ func (ep *endpoint) Delete() error {
 	ep.Lock()
 	epid := ep.id
 	name := ep.name
-	sb, _ := n.getController().SandboxByID(ep.sandboxID)
-	if sb != nil {
-		ep.Unlock()
+	sbid := ep.sandboxID
+	locator := ep.locator
+	ep.Unlock()
+
+	if force {
+		if err = n.validateForceDelete(locator); err != nil {
+			return fmt.Errorf("unable to force delete endpoint %s : %v", name, err)
+		}
+	}
```

Comment:
I understand what you mean. But, I would prefer to retain the `ep.ReleaseAddress` as the last step even for the case of forced delete. my main concern is the fact that if we release the address when the endpoint is still present, that would result in another endpoints potentially grabbing the  address. Even more, when this old endpoint actually goes away (say, `ep.deleteEndpoint()` succeeds at a later time since the remote plugin is activated and hence `n.driver()` works), then we will release the ip-address that is actually being used by another endpoint.
Since with `-f` we are already trying the best to perform a cleanup, if that fails for some reason, I guess it is safe to retain the address than releasing it. WDYT ?